ios/brave-ios/Sources/Brave/Frontend/Browser/TabManager.swift

```diff
@@ -859,13 +862,31 @@ class TabManager: NSObject {
 
   @MainActor private func forgetData(for url: URL, in tab: Tab) async {
     guard let etldP1 = url.baseDomain else { return }
+
     // Start a task to delete all data for this etldP1
     // The task may be delayed in case we want to cancel it
     let dataStore = tab.webView?.configuration.websiteDataStore
     await dataStore?.deleteDataRecords(
       forDomain: etldP1
     )
 
+    // Delete the history for forgotten websites
+    if let historyAPI = self.historyAPI {
+      let nodes = await historyAPI.search(
+        withQuery: etldP1,
+        options: HistorySearchOptions(
+          maxCount: 0,
+          hostOnly: false,
+          duplicateHandling: .keepAll,
```

Comment:
Why did you change this to `keepAll` instead of deleting duplicates?

Reply:
Same

Reply:
I think delete duplicates is a bad name (unless I misunderstood). It doesn't actually delete them. It basically doesn't return them. So a better name is perhaps skipDuplicates. In any case since I do want duplicates returned (as i'm deleting history) i need to pass keepAll to get duplicates so I can actually delete them too.